ext/standard/user_filters.c

```diff
@@ -196,8 +206,20 @@ php_stream_filter_status_t userfilter_filter(
 	/* filter resources are cleaned up by the stream destructor,
 	 * keeping a reference to the stream resource here would prevent it
 	 * from being destroyed properly */
-	if (stream_prop) {
-		convert_to_null(stream_prop);
+	if (prop_info) {
+		/* Declared property, find the actual value to clean it up */
+		zval *stream_prop = zend_hash_str_find(Z_OBJPROP_P(obj), "stream", sizeof("stream")-1);
+		if (stream_prop) {
+			/* Use direct manipulation to avoid type checking during cleanup.
+			 * We need to break the resource reference regardless of property type constraints. */
+			convert_to_null(stream_prop);
```

Comment:
This needs to be changed as well, as setting a typed property to null may be incorrect. We should use `Z_OBJHANDLER_P(obj, unset_property)`.

Reply:
Hmm, I wonder if it should be really used. It works on first call, then the property is fully uninitialized. Calling another stream operation breaks because of this (and many tests break when trying to implement it). The current approach "maintains" the property by just setting it to null. I guess we may keep it like this, unless I missed the point?

Reply:
Ah ok, I was asking because you had marked this as resolved :)

Right, unsetting will not work. Thinking more about it, as the property was assigned with a resource above, it implies that the property has  no type hint (or it's mixed). So we can simply assign null here, after all (with the same method as above).